microservices/coco-attestation/uyuni-coco-attestation.spec

```diff
@@ -0,0 +1,126 @@
+#
+# spec file for package uyuni-coco-attestation
+#
+# Copyright (c) 2024 SUSE LLC
+#
+# All modifications and additions to the file contributed by third parties
+# remain the property of their copyright owners, unless otherwise agreed
+# upon. The license for this file, and modifications and additions to the
+# file, is the same license as for the pristine package itself (unless the
+# license for the pristine package is not an Open Source License, in which
+# case the license is the MIT License). An "Open Source License" is a
+# license that conforms to the Open Source Definition (Version 1.9)
+# published by the Open Source Initiative.
+
+# Please submit bugfixes or comments via https://bugs.opensuse.org/
+#
+
+Name:           uyuni-coco-attestation
+Version:        5.0.0
+Release:        0
+Summary:        Uyuni utility for Confidential Computing Attestation
+License:        Apache-2.0
+Group:          System/Daemons
+URL:            https://www.uyuni-project.org
+Source0:        %{name}-%{version}.tar.gz
+BuildRequires:  fdupes
+BuildRequires:  java-devel >= 11
+BuildRequires:  maven-local
+BuildRequires:  mvn(org.uyuni-project:uyuni-java-parent:pom:)
+BuildRequires:  mvn(org.uyuni-project:uyuni-java-common)
+BuildRequires:  mvn(org.apache.commons:commons-ognl)
+BuildRequires:  mvn(org.apache.logging.log4j:log4j-api)
+BuildRequires:  mvn(org.apache.logging.log4j:log4j-core)
+BuildRequires:  mvn(org.postgresql:postgresql)
+BuildRequires:  mvn(org.javassist:javassist)
+BuildRequires:  mvn(org.mybatis:mybatis)
+BuildRequires:  mvn(com.mchange:mchange-commons-java)
+BuildRequires:  mvn(com.mchange:c3p0)
+
+BuildArch:      noarch
+
+%description
+System daemon used by Uyuni to validate the results of confidential computing attestation.
+
+%package module-snpguest
+Summary:        Confidential computing SNPGuest attestation module for Uyuni
+Requires:       snpguest
+
+%description module-snpguest
+Module for the Uyuni Confidential Computing Attestation that uses SnpGuest.
+
+%package        javadoc
+Summary:        API documentation for %{name}
+
+%description    javadoc
+Package containing the Javadoc API documentation for %{name}.
+
+%prep
+%setup -q
+
+# Disable the common module it will be provided by the installed dependency
+%pom_disable_module '../uyuni-java-common'
+
+# Make sure there are not dependencies on the modules projects
+%pom_remove_dep 'org.uyuni-project.coco-attestation.module:' attestation-core/pom.xml
```

Comment:
Is this line removing the dependency for the core module only or for all?

Reply:
It's only for  `attestation-core`. This lines make sure the maven pom for attestation-core.jar does not declare any dependency on a module implementation. This is because if the pom contains such dependency, the `uyuni-coco-attestation` package we create in OBS will require the corresponding module package, and we don't want that, as the modules should be pluggable in any combination (it should be the module that requires the core, not the opposite).

On the other hand, during development it's convenient to declare such runtime dependency, otherwise by default you won't have any module on the classpath when you run the app through an IDE. I could probably define a configuration for the maven exec plugin to achieve the same though. I will look into that.